### ISSUE TITLE
Runtime error when converting large CBE to CTE

### ISSUE THREAD

**Issue by kengruven** [kstenerud/enctool#8]
I made a big JSON file using an online tool to create some nonsense data, and used enctool to convert it to CTE to play around with.

I tried converting this CTE to CBE and back to CTE (a quick way to pretty-print CTE, since enctool doesn't support direct CTE-CTE conversion), and it gave an error on the CBE->CTE step.

```
$ ./enctool convert -s crash.cte -df cbe|./enctool convert -df cte
[...many lines of output, then...]
        address = runtime error: slice bounds out of range [:4171] with capacity 4096
$
```

I cropped the top/bottom of the file, as much as possible, while still retaining the error.  This file is now only about 10KB (280 lines).

[crash.cte.gz](https://github.com/kstenerud/enctool/files/6127299/crash.cte.gz)

**Comment by kstenerud** [comment]
Yeah, the CBE codec is still using the old reader code. I've retired it and moved CTE over, but haven't had time to do the CBE side yet (next on my list after finishing off NA and Resource ID with concatenation). It uses a buffer size of 4096, and has issues with larger data structures.
